src/tensorlake/documentai/client.py

```diff
@@ -394,256 +436,236 @@ def delete_file(self, file_id: str):
+        page_range: Optional[str] = None,
+        labels: Optional[dict] = None,
+        mime_type: Optional[MimeType] = None,
+        wait_for_completion: bool = False,
+    ) -> Union[str, ParseResult]:
+        dataset_parse_req = self._create_dataset_parse_req(
+            file,
+            page_range,
+            labels,
+            mime_type,
         )
-        response.raise_for_status()
-        return Job.model_validate(response.json())
 
-    async def get_job_async(self, job_id: str) -> Job:
-        """
-        Get the result of a job by its ID asynchronously.
-        """
-        client = httpx.AsyncClient(base_url=DOC_AI_BASE_URL, timeout=None)
-        response = await client.get(
-            url=f"jobs/{job_id}",
-            headers=self.__headers__(),
+        response = await self._arequest(
+            "POST",
+            f"/datasets/{dataset.slug}/parse",
+            json=dataset_parse_req,
         )
-        response.raise_for_status()
-        return Job.model_validate(response.json())
+        response_json = response.json()
 
-    def delete_job(self, job_id: str):
-        """
-        Delete a job by its ID.
-        """
-        asyncio.run(self.delete_job_async(job_id))
+        if not wait_for_completion:
+            return response_json["parse_id"]
 
-    async def delete_job_async(self, job_id: str):
-        """
-        Delete a job by its ID asynchronously.
-        """
-        client = httpx.AsyncClient(base_url=DOC_AI_BASE_URL, timeout=None)
-        response = await client.delete(
-            url=f"jobs/{job_id}",
-            headers=self.__headers__(),
-        )
-        response.raise_for_status()
+        return await self.wait_for_completion_async(response_json["parse_id"])
 
-    def jobs(self, cursor: Optional[str] = None) -> PaginatedResult[Job]:
+    def _create_dataset_parse_req(
+        self,
+        file: str,
+        page_range: Optional[str] = None,
+        labels: Optional[dict] = None,
+        mime_type: Optional[MimeType] = None,
+    ) -> Dict[str, Any]:
         """
-        Get a list of jobs.
+        Create a request body for parsing a file in a dataset.
         """
-        return asyncio.run(self.jobs_async(cursor))
+        payload = {}
 
-    async def jobs_async(self, cursor: Optional[str] = None) -> PaginatedResult[Job]:
-        """
-        Get a list of jobs asynchronously.
-        """
-        client = httpx.AsyncClient(base_url=DOC_AI_BASE_URL, timeout=None)
-        response = await client.get(
-            url="/jobs",
-            headers=self.__headers__(),
-            params={"cursor": cursor} if cursor else None,
-        )
-        response.raise_for_status()
-        result = PaginatedResult[Job].model_validate(response.json())
-        return result
+        # file check
+        if file.startswith("http://") or file.startswith("https://"):
+            payload["file_url"] = file
+        elif file.startswith("tensorlake-"):
```

Comment:
This makes sense for now, but I think this is something we need to update in the future. For example, what if I sent:
```
file="tensorlake-contract.pdf"
```

Then would this assume that this is a file_id?

Reply:
TODO:
1. @drguthals Ensure docs specify that Datasets is not uploading documents and so this needs to be either a URL, FileID, or Raw Test
2. @miguelhrocha Ensure that this is included in the test cases and that when someone does do this we will error out saying that `file_id 'tensorlake-contract.pdf' is not valid` (or whatever the error message says)